tests/test_user_panel.py

```diff
@@ -323,39 +344,133 @@ def test_assigning_resets_tickets(db, user_client):
-    assert False
+    assert response.status_code == 200
+    assert template_used(response, "ep19/bs/user_panel/profile_settings.html")
+    assert user.first_name in response.content.decode()
+    assert user.last_name in response.content.decode()
+    assert user.email in response.content.decode()
+    assert p3_profile.tagline in response.content.decode()
+    assert p3_profile.twitter in response.content.decode()
+    assert attendee_profile.getBio().body in response.content.decode()
+
+
+def test_profile_settings_updates_user_data(user_client):
+    url = reverse('user_panel:profile_settings')
+    payload = dict(
+        first_name='One',
+        last_name='Two',
+        email='one@two.three',
+        tagline='I am the one',
+        twitter='one',
+        bio='One to the Two',
+    )
 
+    response = user_client.post(url, data=payload)
 
-@pytest.mark.xfail
-def test_profile_settings_updates_attendee_profile_data():
-    assert False
+    assert response.status_code == 200
+    assert template_used(response, "ep19/bs/user_panel/profile_settings.html")
+    assert payload['first_name'] in response.content.decode()
+    assert payload['last_name'] in response.content.decode()
+    assert payload['email'] in response.content.decode()
+    assert payload['tagline'] in response.content.decode()
+    assert payload['twitter'] in response.content.decode()
+    assert payload['bio'] in response.content.decode()
+
+
+def test_profile_settings_forbids_using_registered_email(user_client):
+    url = reverse('user_panel:profile_settings')
+    user = user_client.user
+    original_email = user.email
+    another_user = make_user()
+    payload = dict(
+        email=another_user.email,
+    )
 
+    response = user_client.post(url, data=payload)
 
-@pytest.mark.xfail
-def test_profile_settings_updates_p3_profile_data():
-    assert False
+    assert response.status_code == 200
+    assert template_used(response, "ep19/bs/user_panel/profile_settings.html")
+    user.refresh_from_db()
+    assert user.email == original_email
 
 
-@pytest.mark.xfail
-def test_profile_settings_updates_image_settings():
+def test_profile_settings_updates_image_settings(user_client):
     """
     4 scenarios to test - show no image, show gravatar, show url, show image.
     """
-    assert False
+    url = reverse('user_panel:profile_settings')
+    user = user_client.user
+    attendee_profile = user.attendeeprofile
+    p3_profile = attendee_profile.p3_profile
+    required_fields = {
+        "first_name": user.first_name,
+        "last_name": user.last_name,
+        "email": user.email,
+    }
+
+    # Show no image
+    response = user_client.post(url, data=required_fields)
+
+    assert response.status_code == 200
+    attendee_profile.refresh_from_db()
+    p3_profile.refresh_from_db()
+    assert not attendee_profile.image
+    assert p3_profile.image_url == ""
+    assert p3_profile.image_gravatar == False
+
+    # Provide image url
```

Comment:
I feel like those should ideally be four separate tests. Looks like it's a good case for @parametrize as well

Reply:
That's a good idea, going to fix that.

Reply:
When looking at the test decided against using parametrize - I think it would have been less readable.